__CONTEXT__
<br>In this exercise, you will use the same data set about forest fires in Brazil from Module 3.

Forest fires in Brazil are occurring more frequently and becoming more severe every year. Forestry officials need to understand the frequency of the forest fires and find other patterns using exploratory data analysis to help them prevent these destructive fires. For this purpose, you will analyse forest fire data from Brazilian forestry departments.

You can download the data set here:
<br>https://dados.gov.br/dataset/sistema-nacional-de-informacoes-florestais-snif/resource/949310f1-05bc-4f56-a63f-aef67aac6164


For this exercise, the data has been pre-formatted as comma-separated values and translated into English where applicable. The data is available as a csv file in the folder named as __dataset__. Make sure that the dataset folder and Python Notebook are at the same level. 

<br>The variables in the data set are:
-	Year: when the fire occured
-	State: where the fire was reported
-	Month: when the fire occurred
-	Number of fires: frequency reported
-	Date reported: when the fire was reported.

## TO DO - Activity 1 - Import Packages
In the code cell below import the 
- Pandas as __pd__ 
- NumPy as __np__
- pyplot from matplotlib as __plt__.

In [2]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

As a next step, we will import the csv data into the Pandas DataFrame. You have already learnt how to import the csv data from a file into Pandas DataFrames.

# TO DO - Activity 2 - Load Data from CSV File to Pandas DataFrame
In this step, you will write the code to load a CSV file into a Pandas DataFrame.

Write code to perform the following:
1. Create a DataFrame named df_ffire.
2. In this DataFrame, load a csv file using the Pandas __read_csv()__ function.
3. The full path and filename is __"./dataset/module4/brazilianfire.csv"__.
4. Note that the CSV file has a header record with variables. You need to rearrange the data in DataFrame in the this order:
    - Date Reported
    - Year
    - Month
    - State
    - Number of Fires
5. Display the DataFrame (top 10 rows) and verify that the columns are in order.
6. Generate the statistical summary of the dataFrame for all the columns.

In [28]:
names=['Date Reported','Year','Month','State','Number of Fires']
df_ffire = pd.read_csv("./dataset/brazilian_fire.csv")[names]
df_ffire.head(10)

Date Reported  Year    Month State  Number of Fires
0     1/01/1998  1998  January  Acre              0.0
1     1/01/1999  1999  January  Acre              0.0
2     1/01/2000  2000  January  Acre              0.0
3     1/01/2001  2001  January  Acre              0.0
4     1/01/2002  2002  January  Acre              0.0
5     1/01/2003  2003  January  Acre             10.0
6     1/01/2004  2004  January  Acre              0.0
7     1/01/2005  2005  January  Acre             12.0
8     1/01/2006  2006  January  Acre              4.0
9     1/01/2007  2007  January  Acre              0.0

In [29]:
df_ffire.describe(include='all')

Date Reported         Year    Month    State  Number of Fires
count           6454  6454.000000     6454     6454      6454.000000
unique            20          NaN       12       27              NaN
top        1/01/1998          NaN  January  Alagoas              NaN
freq             324          NaN      541      240              NaN
mean             NaN  2007.461729      NaN      NaN       108.293163
std              NaN     5.746654      NaN      NaN       190.812242
min              NaN  1998.000000      NaN      NaN         0.000000
25%              NaN  2002.000000      NaN      NaN         3.000000
50%              NaN  2007.000000      NaN      NaN        24.000000
75%              NaN  2012.000000      NaN      NaN       113.000000
max              NaN  2017.000000      NaN      NaN       998.000000

# TO DO - Activity 3

Based on the information collected so far, can you answer these questions?

1. What is the maximum number of fires reported on any given day?
2. In the dataset, what is the total number of states that have reported fire incidents?

## Provide the Answers to Activity 3 in this Cell
__Questions__
1. What is the maximum number of fires reported on any given day?
2. In the dataset, what is the total number of states that have reported fire incidents?

__Answers__
1. 998
2. 27

### NOTE: We will extend this notebook further, through the assignment in Module 5.